### PR TITLE
Support FixedLenByteArrayType in DeltaByteArrayEncoder

### LINKED ISSUE #95: Support FixedLenByteArrayType in DeltaByteArrayEncoder
See parquet-mr: https://github.com/apache/parquet-mr/blob/master/parquet-column/src/main/java/org/apache/parquet/column/values/factory/DefaultV2ValuesWriterFactory.java#L81

### PR DESCRIPTION
**Pull request by sunchao**
This adds support for FixedLenByteArrayType in DeltaByteArrayEncoder.
The implementation basically reuses the code from ByteArrayType.

Fixes #95.

### CODE REVIEW

File: src/encodings/decoding.rs [hunk 1]
```diff
@@ -815,13 +819,13 @@ impl<> Decoder<ByteArrayType> for DeltaByteArrayDecoder<ByteArrayType> {
     assert!(self.suffix_decoder.is_some());
 
     let num_values = cmp::min(buffer.len(), self.num_values);
+    let mut v: [ByteArray; 1] = [ByteArray::new(); 1];
```
Comment (sadikovi) [inline src/encodings/decoding.rs]: 👍

File: src/encodings/decoding.rs [hunk 2]
```diff
@@ -842,6 +846,17 @@ impl<> Decoder<ByteArrayType> for DeltaByteArrayDecoder<ByteArrayType> {
   }
 }
 
+impl<> Decoder<FixedLenByteArrayType> for DeltaByteArrayDecoder<FixedLenByteArrayType> {
+  fn set_data(&mut self, data: ByteBufferPtr, num_values: usize) -> Result<()> {
+    let s: &mut DeltaByteArrayDecoder<ByteArrayType> = unsafe { mem::transmute(self) };
```
Comment (sadikovi) [inline src/encodings/decoding.rs]: That is an interesting approach!

**Comment by sunchao** [comment]
Hmm yes. It happened again. Will investigate.

**Comment by sunchao** [comment]
Merged. Thanks @sadikovi !

**Pull request merged**